ProcessMaker/Http/Controllers/Api/UserController.php

```diff
@@ -309,14 +310,14 @@ public function updateGroups(User $user, Request $request)
                 if (!is_array($groups)) {
                     $groups = array_map('intval', explode(',', $request->groups));
                 }
-                $user->groups()->sync($groups);
+                $data = $user->groups()->sync($groups);
             } else {
                 $user->groups()->detach();
             }
         } else {
             return response([], 400);
         }
-
+        event(new UserGroupMembershipUpdated($data, $user));
```

Comment:
When there are no groups and the groups are detached we get into a bug situation as `$data` is undefined.